src/result/result.py

```diff
@@ -62,10 +61,12 @@ def __ne__(self, other: Any) -> bool:
     def __hash__(self) -> int:
         return hash((True, self._value))
 
-    def is_ok(self) -> bool:
+    @staticmethod
```

Comment:
`staticmethod` is mostly a historical ‘mistake’ in python. i don't see a reason to add it here (and in various other places)

Reply:
Why do you feel that it is a mistake?

Reply:
https://mail.python.org/pipermail/python-ideas/2016-July/041189.html 🙃

Reply:
I get that Guido is a figure of authority, but I don't feel like this quote adds much information to the conversation. I think I'm making a fair argument as to why the `staticmethod` decorator is appropriate here, and nothing in the quote (or thread) seems to be contradicting the arguments I've put forward.

My guess is that Guido felt it was a mistake because he, in hindsight, didn't want to encourage the _concept_ of a static method. That doesn't change the fact that once we have this decorator, it is useful in cases where we have static methods that won't be going away. Clearly I'm speculating here, but there isn't a lot to go by.

If you still feel this is controversial, but you like the other changes I made, I can always separate out the decorator addition into its own PR and we can discuss it further there.

Reply:
@wbolster The change to `staticmethod` has been reverted for now.